rasterio/_io.pyx

```diff
@@ -1749,7 +1749,31 @@ cdef class RasterUpdater(RasterReader):
         else:
             retval = io_ubyte(
                 hmask, 1, xoff, yoff, width, height, mask)
-        
+
+    def build_overviews(self, factors, resampling=Resampling.nearest):
+        """Build overviews at one or more decimation factors for all
+        bands of the dataset."""
+        cdef int *factors_c = NULL
+        cdef const char *resampling_c = NULL
+
+        if self._hds == NULL:
+            raise ValueError("can't write closed raster file")
+
+        # Allocate arrays.
+        if factors:
+            factors_c = <int *>_gdal.CPLMalloc(len(factors)*sizeof(int))
+            for i, factor in enumerate(factors):
+                factors_c[i] = factor
+
+        with cpl_errs:
```

Comment:
lines 1768-1775 should be indented further, or it looks like you risk a null pointer being passed in for ```factors_c``` input to ```GDALBuildOverviews()```